app/config/firebase.ts

```diff
@@ -0,0 +1,89 @@
+import { initializeApp } from 'firebase/app';
+import {
+  type Auth,
+  type User,
+  getAuth,
+  signInWithRedirect,
+  GoogleAuthProvider,
+  GithubAuthProvider,
+} from 'firebase/auth';
+import type { FirebaseConfig } from '../types/firebase';
+import { getFirestore, doc, getDoc, setDoc } from 'firebase/firestore';
+
+interface ReloadUser {
+ screenName: string
+}
+
+interface ExtendedUser extends User {
+  reloadUserInfo?: ReloadUser;
+}
+
+const firebaseConfig: FirebaseConfig = {
+  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY!,
+  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN!, //TODO: change to app domain
+  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID!,
+  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MSG_SENDER_ID,
+  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID!,
+};
+
+if (
+  !firebaseConfig.apiKey ||
+  !firebaseConfig.authDomain ||
+  !firebaseConfig.projectId ||
+  !firebaseConfig.appId
+) {
+  throw new Error(
+    'Firebase configuration error: Missing essential configuration values'
+  );
+}
+//initialize firebase app
+const firebaseApp = initializeApp(firebaseConfig);
+
+export const auth: Auth = getAuth(firebaseApp); //initialize Firebase Authentication service
+auth.useDeviceLanguage(); //detecting/using user perferrd languague on their devices
+export const db = getFirestore(firebaseApp); //initialize Firestore service
+
+const googleProvider = new GoogleAuthProvider();
+const ghProvider = new GithubAuthProvider();
+googleProvider.setCustomParameters({
+  prompt: 'select_account',
+});
+
+export const signUpWithGoogle = () => signInWithRedirect(auth, googleProvider);
+export const signUpWithGitHub = () => signInWithRedirect(auth, ghProvider);
+
+
+export const createUserDoc = async (userAuth: ExtendedUser) => {
+  const userDocRef = doc(db, 'users', userAuth.uid);  //create a User Document reference
+
+  console.log('userDocRef:', userDocRef);
```

Comment:
since this will be imported by a frontend component, I think it's best to remove this console.log so we're not logging to the user's client 😄